apps/playoff-schedule/frontend/src/App.tsx

```diff
@@ -0,0 +1,419 @@
+
+    return (timeA > matchTimeMissing && timeB > matchTimeMissing) ? timeA - timeB
+      : (timeA === matchTimeMissing && timeB > matchTimeMissing) ? sortABeforeB
+      : (timeA > matchTimeMissing && timeB === matchTimeMissing) ? sortBBeforeA
+      : (weightA !== weightB) 
+        ? weightA - weightB 
+        : a.match_number - b.match_number;
+  }, []);
+
+  const performSearch = useCallback(async (eventKey: string) => {
+    if (!eventKey) return;
+    
+    resetInSearch();
+    setActiveEventKey(eventKey);
+
+    try {
+      const teamsUrl = urlTeamsInEvent(eventKey);
+      const matchesUrl = urlMatches(eventKey);
+      const rankingsUrl = urlRankings(eventKey);
+
+      const [teamsData, matchesData, rankingsData] = await Promise.all([
+        fetchFromProxy<TeamsInEventType[]>(teamsUrl),
+        fetchFromProxy<MatchesSimpleType[]>(matchesUrl),
+        fetchFromProxy<RankingsResponse>(rankingsUrl).catch(() => ({ rankings: [] }))
+      ]);
+
+      Array.isArray(teamsData) && Array.isArray(matchesData)
+        ? (() => {
+            setTeams(teamsData);
+            matchesData.sort(sortMatches);
+            setAllMatches(matchesData);
+            setTeamRank(rankingsData.rankings.find(r => r.team_key === targetTeamKey) ?? null);
+            setSearchStatus("success");
+          })()
+        : setSearchStatus("error");
+
+    } catch {
+      setSearchStatus("error");
+    }
+  }, [setActiveEventKey, sortMatches]);
+
+  const handleSearchSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    const formattedKey = inputEventKey.trim().toLowerCase();
+    void performSearch(formattedKey);
+  };
+
+  useEffect(() => {
+    if (activeEventKey && searchStatus === "idle") {
+        void performSearch(activeEventKey);
+    }
+  }, [activeEventKey, performSearch, searchStatus]);
+
+  useEffect(() => {
+    const intervalId: number | undefined = searchStatus === "success" && activeEventKey ? 
+      window.setInterval(() => {
+        const matchesUrl = urlMatches(activeEventKey);
+        const rankingsUrl = urlRankings(activeEventKey);
+
+        void fetchFromProxy<MatchesSimpleType[]>(matchesUrl).then(data => {
+           if (Array.isArray(data)) {
+             data.sort(sortMatches);
+             setAllMatches(data);
+           }
+        }).catch(console.error);
+
+        void fetchFromProxy<RankingsResponse>(rankingsUrl).then(data => {
+            const myRank = data.rankings.find(r => r.team_key === targetTeamKey) ?? null;
+            setTeamRank(myRank);
+        }).catch(console.error);
+
+      }, refreshIntervalMs) : undefined;
+
+    return () => {
+      if (intervalId !== undefined) {
+        window.clearInterval(intervalId);
+      }
+    };
+  }, [activeEventKey, searchStatus, sortMatches]);
+
+  const teamNameMap = useMemo(() => {
+    const map: Map<string, string> = new Map();
+    teams.forEach(team => map.set(team.key, team.nickname));
```

Comment:
consider using a `Record<string,string>` instead of a map.
```suggestion
    const map: Map<string, string> = {};
    teams.forEach(team => {map[team.key] = team.nickname;});
```